### PR TITLE
ZDM-159 Add README

### PR DESCRIPTION
**Pull request by alicel**
Also removed ZDM Util's separate README that is superseded by the main one.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,2 +1,123 @@
-# zdm-proxy-automation
-Automation suite to deploy the Zero Downtime Migration Proxy (ZDM Proxy)
+# ZDM Proxy Automation
+
+### _An automation suite to deploy and manage the Zero Downtime Migration Proxy_
```
Comment (joao-r-reis) [inline README.md:3]: I'd probably add a link to the proxy repo here: "Zero Downtime Migration Proxy" even though there's already one in the overview section.

**Review by joao-r-reis — APPROVED** [review]
👍

File: README.md (hunk 1 above)
Comment (alicel) [reply README.md]: Done

File: README.md [hunk 2]
```diff
@@ -1,2 +1,123 @@
...
+* The variables with which the script has been executed.  If you need to tear down all the infrastructure provisioned, you can go to the directory of the root module that you used and execute the command `terraform destroy` followed by all these variables (these are also persisted to file `tf_latest_vars.txt` in this directory).
+* The output of the automation, with the IP addresses of the instances that were provisioned and the ZDM VPC ID. This output is also saved to file `zdm_output.txt` in the `zdm-proxy-automation` base directory.
+
+In addition, the Terraform automation creates the following two files:
+* `zdm_ansible_inventory`, which contains the newly provisioned instance IPs in a file that can be passed directly to the Ansible automation to indicate which instances it must operate on.
+* `zdm_ssh_config`, a custom SSH configuration file that allows you to easily connect to the jumphost or ZDM Proxy instances from an external machine with a single command. See [here](https://docs.datastax.com/en/astra-serverless/docs/migrate/deployment-infrastructure.html#_connecting_to_the_zdm_infrastructure_from_an_external_machine) for details.
+
+## Docker Compose deployment
```
Comment (grighetto) [inline README.md:83]: I'd change this to `Docker Compose for Local Development`. The goal of the docker-compose definition is mostly development and integration testing (the plan is to have NoSQLBench verifying the Ansible setup too). The term `deployment` makes me think it can be used for actual deployment, which is not the case. Even though it's somewhat clarified in the first paragraph, it's good to call it out in the title.

Also, this would probably be a better fit in a CONTRIBUTING or DEVELOPERS section,  instead of the general purpose README.

File: README.md [hunk 3]
```diff
@@ -1,2 +1,123 @@
...
+
+The Compose file `docker_compose.yml` in the base directory of this repository and the scripts in the `compose` directory can be used to spin up a local ZDM deployment using containers. This can be convenient for exploration and local testing, including testing of the Ansible automation.
+
+The ZDM deployment thus created includes:
+* A proxy network of type `bridge`.
+* Two single-node Cassandra clusters, with specified versions, which are used as Origin and Target.
+* A jumphost machine, from which the Ansible automation is run (specifically, the playbook to deploy the ZDM Proxy).
```
Comment (grighetto) [inline README.md:90]: I'd use `container` instead of `machine` here

File: README.md [hunk 4]
```diff
@@ -1,2 +1,123 @@
...
+* Three Docker containers representing three Linux machines, which are used to run the ZDM Proxy instances as Docker containers (using Docker-in-Docker).
```
Comment (grighetto) [inline README.md:91]: IMO this line is a bit confusing. A simpler suggestion: `3 Ubuntu containers to host each of the ZDM Proxy instances`.

File: README.md [hunk 5]
```diff
@@ -1,2 +1,123 @@
...
+
+As prerequisites, ensure that:
+* [Docker](https://docs.docker.com/get-docker/) and [Docker Compose](https://docs.docker.com/compose/install/) are installed on your machine 
+* The Docker daemon is up and running
+
+You will be using the `docker compose` command, which in some installations may be available as `docker-compose`.
+
+To create this deployment, simply run:
+```shell
+docker compose up
```
Comment (grighetto) [inline README.md:102]: The command is `docker-compose`

File: README.md [hunk 6]
```diff
@@ -1,2 +1,123 @@
...
+```
+
+To view the logs:
+```shell
+docker compose logs
```
Comment (grighetto) [inline README.md:107]: By the default, docker-compose will print the logs on the screen once you start it (unless you start it in detached mode, but we're not suggesting that above). Anyway, we can keep it here or simply reference the official docs below for more info.

File: README.md [hunk 7]
```diff
@@ -1,2 +1,123 @@
...
+```
+
+For more details on Docker Compose, please refer to the [official documentation](https://docs.docker.com/get-started/08_using_compose/).
+
+To issue CQL requests through the proxy, create a shell on the client container (`zdm-proxy-automation_client_1`)
+
+```shell
+docker exec -it zdm-proxy-automation_client_1 bash
```
Comment (grighetto) [inline README.md:115]: I believe we can do it in a single command: `docker exec -it zdm-proxy-automation_client_1 cqlsh zdm-proxy-automation_proxy_1`.
That will take people to a cqlsh session that they can use interactively, I think that's preferable.

**Review by grighetto — COMMENTED** [review]
Made some suggestions to the docker-compose section

File: README.md (hunk 3 above)
Comment (alicel) [reply README.md:90]: done

File: README.md (hunk 4 above)
Comment (alicel) [reply README.md:91]: yes this reads better - however i would like to point out here that the proxy instance themselves also run as docker containers It might make it easier for someone who wants to better understand the setup).

i'd reword it as:

`Three Ubuntu containers to host each ZDM Proxy instance, itself running as a container using Docker-in-Docker`

How does it sound to you?

File: README.md (hunk 5 above)
Comment (alicel) [reply README.md:102]: apparently it depends on the versions and platforms (see [here](https://docs.docker.com/compose/gettingstarted/#step-8-experiment-with-some-other-commands) for example)

File: README.md (hunk 6 above)
Comment (alicel) [reply README.md:107]: Cool, I'll take it out.

File: README.md (hunk 7 above)
Comment (alicel) [reply README.md:115]: done.

File: README.md (hunk 4 above)
Comment (grighetto) [reply README.md:91]: > > Three Ubuntu containers to host each ZDM Proxy instance, itself running as a container using Docker-in-Docker

> How does it sound to you?

Sounds good 👍

**Review by grighetto — APPROVED** [review]
LGTM

File: README.md (hunk 5 above)
Comment (alicel) [reply README.md:102]: i'll change it to docker-compose and mention that on some installations it may be docker compose without hyphen
